components/script/navigation.rs

```diff
@@ -0,0 +1,222 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+//! The listener that encapsulates all state for an in-progress document request.
+//! Any redirects that are encountered are followed. Whenever a non-redirect
+//! response is received, it is forwarded to the appropriate script thread.
+
+use std::cell::Cell;
+
+use base::cross_process_instant::CrossProcessInstant;
+use base::id::{BrowsingContextId, PipelineId, TopLevelBrowsingContextId};
+use content_security_policy::Destination;
+use crossbeam_channel::Sender;
+use http::header;
+use ipc_channel::ipc;
+use net::http_loader::{set_default_accept_language, DOCUMENT_ACCEPT_HEADER_VALUE};
```

Comment:
Can we move these into components/shared/net and remove the net dependency?

Reply:
Yes, sure thing. I was a bit suspicious of this new dependency as well.